lib/services/search_service.dart

```diff
@@ -0,0 +1,180 @@
+import 'package:dio/dio.dart';
+import 'package:logging/logging.dart';
+import 'package:photos/core/configuration.dart';
+import 'package:photos/core/event_bus.dart';
+import 'package:photos/core/network.dart';
+import 'package:photos/db/files_db.dart';
+import 'package:photos/events/local_photos_updated_event.dart';
+import 'package:photos/models/collection.dart';
+import 'package:photos/models/collection_items.dart';
+import 'package:photos/models/file.dart';
+import 'package:photos/models/location.dart';
+import 'package:photos/models/search/location_misc./place_and_bbox.dart';
+import 'package:photos/models/search/location_misc./results_to_list_of_place_and_bbox.dart';
+import 'package:photos/models/search/location_search_result.dart';
+import 'package:photos/services/collections_service.dart';
+import 'package:photos/services/user_service.dart';
+
+class SearchService {
+  List<File> _cachedFiles;
+  Future<List<File>> _future;
+  final _dio = Network.instance.getDio();
+  final _config = Configuration.instance;
+  final _logger = Logger((UserService).toString());
+  final _collectionService = CollectionsService.instance;
+  static const _maximumResultsLimit = 20;
+
+  SearchService._privateConstructor();
+  static final SearchService instance = SearchService._privateConstructor();
+
+  Future<void> init() async {
+    // Intention of delay is to give more CPU cycles to other tasks
+    Future.delayed(const Duration(seconds: 5), () async {
+      // In case home screen loads before 5 seconds and user starts search, future will not be null
+      _future == null
+          ? getAllFiles().then((value) {
+              _cachedFiles = value;
+            })
+          : null;
+    });
+
+    Bus.instance.on<LocalPhotosUpdatedEvent>().listen((event) {
+      _cachedFiles = null;
+      getAllFiles();
+    });
+  }
+
+  Future<List<File>> getAllFiles() async {
+    if (_cachedFiles != null) {
+      return _cachedFiles;
+    }
+    if (_future != null) {
+      return _future;
+    }
+    _future = _fetchAllFiles();
+    return _future;
+  }
+
+  Future<List<File>> _fetchAllFiles() async {
+    _cachedFiles = await FilesDB.instance.getAllFilesFromDB();
+    return _cachedFiles;
+  }
+
+  Future<List<File>> getFilesOnFilenameSearch(String query) async {
+    final List<File> matchedFiles = [];
+    final List<File> files = await getAllFiles();
+    for (int i = 0;
+        (i < files.length) && (matchedFiles.length < _maximumResultsLimit);
+        i++) {
+      File file = files[i];
+      if (file.title.contains(RegExp(query, caseSensitive: false))) {
```

Comment:
Could extract `RegExp(query, caseSensitive: false)` to a separate variable outside the loop so that it doesn't have to be instantiated each time?